gen/builders/generator.go

```diff
@@ -73,15 +79,10 @@ const (
 var GenscriptCommit = "dirty"
 
 // genData is the generation data to stamp into vectors.
-var genData = []schema.GenerationData{
-	{
-		Source:  "genscript",
-		Version: GenscriptCommit,
-	},
-}
+var genData []schema.GenerationData
 
 func init() {
-	genData = append(genData, getBuildInfo()...)
+	genData = getBuildInfo()
```

Comment:
I think we need to refactor this a little bit more. Rather than having a global that makes implicit assumptions about how things are called, I think we can make this a lot more failsafe/tolerant and specific by doing the following. This will let us get to the actual file/line where the `MessageFunc` or the `TipsetFunc` is located.

1. Get rid of this global here.
2. In `generateOne`, calculate the "boilerplate" gendata (that is, specs-actors and lotus versions, like we do now).
3. [Use reflection](https://stackoverflow.com/questions/10742749/get-name-of-function-using-reflection) to get the program counter of `MessageFunc` or `TipsetFunc`.
4. Use [`runtime.FuncForPC`](https://golang.org/pkg/runtime/#FuncForPC) to get the file line info.
5. Append the above to the gendata.

Reply:
Nice, I didn't know you could get file paths of functions via reflection! I did the simplest thing that would work but I'm happy to iterate 😄. I think this will be really useful for consumers.